File: app/src/main/java/com/vitaliyhtc/dagger2investigation/global/di/AppModule.java
package com.vitaliyhtc.dagger2investigation.global.di;

import android.content.Context;

import com.vitaliyhtc.dagger2investigation.global.Dagger2InvestigationApp;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    @Provides
    @Singleton
    Context provideContext(Dagger2InvestigationApp application) {
        return application.getApplicationContext();
    }
}

File: app/src/main/java/com/vitaliyhtc/dagger2investigation/global/di/AppModule.kt
package com.vitaliyhtc.dagger2investigation.global.di

import android.content.Context
import com.vitaliyhtc.dagger2investigation.global.App
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule {

    @Provides
    @Singleton
    fun provideContext(application: App): Context = application.applicationContext

}